src/transformers/models/gpt_neox/modeling_gpt_neox.py

```diff
@@ -258,7 +254,7 @@ def _attn(self, query, key, value, attention_mask=None, head_mask=None):
             query,
             key.transpose(1, 2),
             beta=1.0,
-            alpha=(torch.tensor(1.0, dtype=self.norm_factor.dtype, device=self.norm_factor.device) / self.norm_factor),
+            alpha=self.norm_factor,
```

Comment:
You are right! 
Sorry for dragging this one, TLDR: 
- the `self.head_size ` is always `self.hidden_size / self.num_attention_heads`. But there's  a check to make sure it is divisible! (So basically it's an int)
- the only way for this to break something is if  `self.head_dim = 65504`. In this case, the previous behaviour had: 
(I know it's a huge head dim 🤣 ) 
```python 
>>> torch.sqrt(torch.tensor(65504, device = "cuda", dtype = torch.float32)).to(torch.float16)
tensor(255.8750, device='cuda:0', dtype=torch.float16)
>>> # vs without casting
>>> torch.sqrt(torch.tensor(65504, device = "cuda", dtype = torch.float32))
tensor(255.9375, device='cuda:0')
```
Should be okay to have this